### PR TITLE
Dropdown: support `.dropdown-item` wrapped in `<li>` tags

### PR DESCRIPTION
**Pull request by cpsievert**
Backports #33634 into v4

### CODE REVIEW

**Comment by XhmikosR** [comment]
Let's wait until the main PR has landed and this one's tests pass.

File: js/tests/unit/tab.js
Comment (alpadev) [inline js/tests/unit/tab.js:367]: I think you need to exclude the `dropdown-toggle` here like `.find('.nav-link:not(.dropdown-toggle)')` to make the test work.

**Review by alpadev — CHANGES_REQUESTED** [review]

**Comment by cpsievert** [comment]
FWIW, the failure on Node 14 doesn't appear to be related to this PR

**Review by alpadev — APPROVED** [review]

**Pull request merged**
